### PR TITLE
Feature/knb2

### LINKED ISSUE #911: feature: add a knob GUI
A knob seems an essential GUI to have around, natively.

The couple of knobs around (flatgui/knob and moonlib/mknob) are pretty much based on iemguis anyway.

### PR DESCRIPTION
**Pull request by Ant1r**
This is a reboot of  #938, targeted and rebased to the `improvement/merge-iemgui` branch.

closes #911
Here is a proposal to add a new knob GUI, associated to "Shift+Ctrl+K" key accelerator.

Originally imported from moonlib/mknob, it has now gained several features, such as:

- optional arc drawing
- configurable angular area
- optional ticks
- removable cursor
- +/- mode

knob

for those wanting to read the long discussions and the preparatory commits (that have been compacted here for clarity), see: #938 #919 #918

### CODE REVIEW

File: tcl/dialog_iemgui.tcl [hunk 1]
```diff
@@ -12,6 +12,22 @@ namespace eval ::dialog_iemgui:: {
...
+    set var_iemgui_guitype [concat iemgui_guitype_$vid]
+    global $var_iemgui_guitype
+
+    set gui_proc [concat $p[eval concat $$var_iemgui_guitype]]
+
+    if {[proc_exists $gui_proc]} {
+        return [$gui_proc $mytoplevel {*}$args]
+    }
+}
+
```
[the remark this answers is not in the document]
Comment (umlaeute) [reply tcl/dialog_iemgui.tcl:32]: of course, an even better approach would be to use arrays:

```tcl
proc ::dialog_iemgui::call_gui_proc {mytoplevel p args} {
  if { [info exists ::dialog_iemgui::guitype($mytoplevel)] == 0 } { return }
  set gui_proc "${p}$::dialog_iemgui::guitype($mytoplevel)"
  if { [info procs $gui_proc] ne {} } {
    $gui_proc $mytoplevel {*}$args
  }
}
```

but practically all the iemgui-dialog variables should use that (so i understand that it is not changed right now)

**Comment by umlaeute** [comment]
sidenote: i've rebased my `gitlab/bugfix/iemgui_dialog` (#1570) onto `improvement/merge-iemgui` (#1633) (come to think of it, i should have done it the other way round), and  this cleans up the iemgui_dialog code quite a bit...

**Comment by Ant1r** [comment]
I see, very nice!
Are you suggesting I should rebase (and simplify accordingly) `feature/knb2` on `bugfix/iemgui_dialog`? Or should I wait until you merge `bugfix/iemgui_dialog` and `improvement/merge-iemgui`?

**Comment by umlaeute** [comment]
`bugfix/iemgui_dialog` and `improvement/merge-iemgui` are merged (well somewhat, they already diverged again :-))

in any case: yes, my suggestion is to rebase `feature/knb2` on `bugfix/iemgui_dialog`.

**Comment by Ant1r** [comment]
OK, I've rewritten this PR basing on `bugfix/iemgui_dialog `.
So I close in favor of #1698.

**Pull request closed without merging**